docs/source/tools/watcher.rst

```diff
@@ -98,3 +98,33 @@ The log file will contain lines such as the following:
 - The slave message ``smsg`` sent from BRISC to the other RISC Vs are all Go ``G``; ``D`` would indicate Done
 - The kernel IDs ``k_ids`` running are ``4`` on BRISC, ``3`` on NCRISC and ``5`` on TRISC; look further down the log file
   to see the names and paths to those kernels
+
+Asserts
+-------
+
+With watcher enabled, asserts can be added to kernel code. Tripping an assert will cause the kernel
+to hang and report which assert was tripped in the watcher log. When the watcher is disabled,
+asserts can be left in the kernel code and will be left out by the compiler. An example of an
+assert and the resulting log message is shown below:
+
+.. code-block:: c++
+
+    #include "debug/assert.h"  // Required in all kernels using watcher asserts
+    #include "debug/status.h"  // Pair the assert with a status to see which assert is tripped
+
+    void kernel_main() {
+        uint32_t a = get_arg_val<uint32_t>(0);
+        uint32_t b = get_arg_val<uint32_t>(1);
+
+        DEBUG_STATUS('A', 'S', 'T', '1');
+        ASSERT(a != b);
+    }
+
+If this assert was tripped, the kernel will hang, and the following message will show up in the
```

Comment:
why a hang and not an exception that ends the program?  why log to a file rather than stderr?

Reply:
We output to both the log file and stderr, corrected this in the documentation.

Reply:
I've updated the docs here to be a bit less confusing, the (host-side) program actually does always get an exception and exit, but the kernel itself will hang if it's tensix and early exit if it's ethernet (because we have to), same implementation as noc sanitization.